airbyte-integrations/connectors/source-iterable/source_iterable/streams.py

```diff
@@ -339,8 +340,14 @@ def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
             yield {"list_id": list_record["id"]}
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        if not self.check_unauthorized_key(response):
-            return []
+        if not response.ok:
+            if not self.check_unauthorized_key(response):
+                return []
+            # Avoid block whole of sync if a slice is broken. Skip current slice on 500 Internal Server Error.
+            # See on-call: https://github.com/airbytehq/oncall/issues/1592#issuecomment-1499109251
+            if response.status_code == codes.INTERNAL_SERVER_ERROR:
```

Comment:
why are we only checking the error code instead of the error message like we do in https://github.com/airbytehq/airbyte/pull/24962/files#diff-fb13a3d03af51a56e778745de0380239a6c760904f7e2abb663af62d00fcbe76R46

Reply:
I prefer to check for status code if possible to avoid guessing error code names